### PR TITLE
syncer: refactor gtid code

### PR DESCRIPTION
**Pull request by GregoryIan**

### CODE REVIEW

**Comment by jomenxiao** [comment]
LGTM

**Comment by holys** [comment]
Have you considered the compatibility for older syncer.meta ?

File: syncer/syncer.go [hunk 1]
```diff
@@ -596,13 +592,14 @@ func (s *Syncer) run() (err error) {
 	go s.printStatus()
 
 	pos := s.meta.Pos()
-	// while meet GTIDEvent, save previous gtid to prevent losing binlog from syncer panicing
+	gs, err := s.meta.GTID()
```
Comment (holys) [inline syncer/syncer.go]: If  enable-gtid == false,  gtid string will be empty, and mysql.ParseMysqlGTIDSet(gtidStr) will get error  `invalid GTID format, must UUID:interval[:interval]`

**Comment by holys** [comment]
This RP need fully test. It can't run at all .

**Comment by GregoryIan** [comment]
It's obvious that something wrong with `github.com/siddontang/go-mysql/mysql`. 
I still not fix it, so this pr is still on the way, add a `WIP` tag

File: syncer/gtid.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	"github.com/siddontang/go-mysql/mysql"
+)
+
+// GTIDSet wraps mysql.MysqlGTIDSet
+// todo: wrap mysql.GTIDSet
+type GTIDSet struct {
+	*mysql.MysqlGTIDSet
+}
+
+func parseGTIDSet(gtidStr string) (GTIDSet, error) {
```
Comment (holys) [inline syncer/gtid.go]: Why `GTIDSet` not use pointer  ?

File: syncer/gtid.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	gs, err := mysql.ParseMysqlGTIDSet(gtidStr)
+	if err != nil {
+		return GTIDSet{}, errors.Trace(err)
+	}
+
+	return GTIDSet{gs.(*mysql.MysqlGTIDSet)}, nil
```
Comment (holys) [inline syncer/gtid.go]: What if  `gs`  asserts to `MariadbGTID` type ?

Comment (GregoryIan) [inline syncer/gtid.go]: notice comments above, we don't support `mariadbGTID` now

File: syncer/gtid.go (hunk 2 above)
Comment (GregoryIan) [inline syncer/gtid.go]: why not? because it's small

**Comment by holys** [comment]
Rest LGTM

**Pull request merged**